widgetry/src/style/button_style.rs

```diff
@@ -0,0 +1,135 @@
+    include_labeled_bytes, ButtonBuilder, Color, ControlState, EventCtx, Key, OutlineStyle,
+    ScreenDims, Style, Widget,
+};
+
+#[derive(Clone)]
+pub struct ButtonStyle {
+    pub fg: Color,
+    pub fg_disabled: Color,
+    pub outline: OutlineStyle,
+    pub bg: Color,
+    pub bg_hover: Color,
+    pub bg_disabled: Color,
+}
+
+impl<'a> ButtonStyle {
+    pub fn btn(&self) -> ButtonBuilder<'a> {
+        let base = ButtonBuilder::new()
+            .label_color(self.fg, ControlState::Default)
+            .label_color(self.fg_disabled, ControlState::Disabled)
+            .image_color(self.fg, ControlState::Default)
+            .image_color(self.fg_disabled, ControlState::Disabled)
+            .bg_color(self.bg, ControlState::Default)
+            .bg_color(self.bg_hover, ControlState::Hovered)
+            .bg_color(self.bg_disabled, ControlState::Disabled);
+
+        if self.outline.0 > 0.0 {
+            base.outline(self.outline, ControlState::Default)
+        } else {
+            base
+        }
+    }
+
+    pub fn text(&self, text: &'a str) -> ButtonBuilder<'a> {
+        self.btn().label_text(text)
+    }
+
+    pub fn icon(&self, image_path: &'a str) -> ButtonBuilder<'a> {
+        icon_button(self.btn().image_path(image_path))
+    }
+
+    pub fn icon_bytes(&self, labeled_bytes: (&'a str, &'a [u8])) -> ButtonBuilder<'a> {
+        icon_button(self.btn().image_bytes(labeled_bytes))
+    }
+
+    pub fn icon_text(&self, image_path: &'a str, text: &'a str) -> ButtonBuilder<'a> {
+        self.btn()
+            .label_text(text)
+            .image_path(image_path)
+            .image_dims(ScreenDims::square(18.0))
+    }
+
+    pub fn dropdown(&self) -> ButtonBuilder<'a> {
+        self.icon_bytes(include_labeled_bytes!("../../icons/arrow_drop_down.svg"))
+            .image_dims(12.0)
+            .stack_spacing(12.0)
+            .label_first()
+    }
+
+    pub fn popup(&self, text: &'a str) -> ButtonBuilder<'a> {
+        self.dropdown().label_text(text)
+    }
+}
+
+impl<'a> Style {
+    /// title: name of previous screen, which you'll return to
+    pub fn btn_back(&self, title: &'a str) -> ButtonBuilder<'a> {
+        self.btn_plain
+            .icon_bytes(include_labeled_bytes!("../../icons/nav_back.svg"))
+            .label_text(title)
+            .padding_left(8.0)
+            .font_size(30)
+    }
+
+    /// A right facing caret, like ">", suitable for paging to the "next" set of results
+    pub fn btn_next(&self) -> ButtonBuilder<'a> {
+        self.btn_plain
+            .icon_bytes(include_labeled_bytes!("../../icons/next.svg"))
+    }
+
+    /// A left facing caret, like "<", suitable for paging to the "next" set of results
+    pub fn btn_prev(&self) -> ButtonBuilder<'a> {
+        self.btn_plain
+            .icon_bytes(include_labeled_bytes!("../../icons/prev.svg"))
+    }
+
+    /// An "X" button to close the current state. Bound to the escape key.
+    pub fn btn_close(&self) -> ButtonBuilder<'a> {
```

Comment:
There were a few methods which *could* have been moved to ButtonStyle but I left the on Style since it lets things be a little more concise and I think we'll always want to use the same underlying button style.

e.g. I think it's better to:

```
ctx.style().close_widget(ctx)
```

Rather than have to:
```
ctx.style().btn_plain.close_widget(ctx)
```

I could be convinced otherwise though... e.g. maybe there's a reason we'll someday want to make an orange "close" button:
```
ctx.style().btn_primary_plain.close_widget(ctx)
```